dev/AppNotifications/AppNotificationTelemetry.h

```diff
@@ -0,0 +1,331 @@
+        }
+    }
+    CATCH_LOG()
+
+    DEFINE_EVENT_METHOD(LogGetAllAsync)(
+        winrt::hresult hr,
+        std::wstring const& appId) noexcept try
+    {
+        if (c_maxEventLimit >= UpdateLogEventCount())
+        {
+            TraceLoggingClassWriteMeasure(
+                "GetAllAsync",
+                TelemetryPrivacyDataTag(PDT_ProductAndServicePerformance),
+                _GENERIC_PARTB_FIELDS_ENABLED,
+                TraceLoggingHexUInt32(hr, "OperationResult"),
+                TraceLoggingWideString(appId.c_str(), "AppId"),
+                TraceLoggingBool(IsPackagedApp(), "IsAppPackaged"),
+                TraceLoggingWideString(GetAppName().c_str(), "AppName"));
+        }
+    }
+    CATCH_LOG()
+
+private:
+    wil::srwlock m_lock;
+    ULONGLONG m_lastFiredTick = 0;
+    UINT m_eventCount = 0;
+
+    static constexpr ULONGLONG c_logPeriod = 1000; // One second
+    static constexpr UINT c_maxEventLimit = 10;
+
+    UINT UpdateLogEventCount()
+    {
+        ULONGLONG currentTick = GetTickCount64();
+
+        auto lock{ m_lock.lock_exclusive() };
+
+        // Only fire limiting events every log period to prevent too many events from being fired
+        if ((currentTick - m_lastFiredTick) > c_logPeriod)
+        {
+            m_eventCount = 0;
+            m_lastFiredTick = currentTick;
+        }
+
+        m_eventCount++;
+
+        return m_eventCount;
+    }
+
+    inline bool IsPackagedApp() const
+    {
+        static const bool isPackagedApp{ AppModel::Identity::IsPackagedProcess() };
+
+        return isPackagedApp;
+    }
+
+    inline const std::wstring& GetAppName() const
+    {
+        static const std::wstring appName{ IsPackagedApp() ? GetAppNamePackaged() : GetAppNameUnpackaged() };
+
+        return appName;
+    }
+
+    std::wstring GetAppNamePackaged() const
+    {
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH]{};
+
+        UINT32 appUserModelIdSize = ARRAYSIZE(appUserModelId);
+        auto result = GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId);
+        if (result != ERROR_SUCCESS)
+        {
+            wcscpy_s(appUserModelId, L"AppUserModelId not found");
+            LOG_WIN32(result);
+        }
+
+        return appUserModelId;
+    }
+
+    std::wstring CensorFilePath(const std::wstring& filepath) const
+    {
+        return { !PathIsFileSpecW(filepath.c_str()) ? PathFindFileNameW(filepath.c_str()) : filepath };
+    }
+
+    std::wstring GetAppNameUnpackaged() const
+    {
+        std::wstring appName;
+
+        wil::unique_cotaskmem_string processName;
+        auto result = wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, processName);
+        if (result == ERROR_SUCCESS)
+        {
+            appName = CensorFilePath(processName.get());
+        }
+        else
+        {
+            appName = L"ModuleFileName not found";
+            LOG_WIN32(result);
```

Comment:
Still only logging instead of throwing. Was this intended?

Reply:
No, that's something I missed. thanks for checking.

It's fixed now